openmc/geometry.py

```diff
@@ -65,14 +63,22 @@ def get_offset(self, path, filter_offset):
 
         """
 
+        # Find the distribcell index of the cell.
+        cells = self.get_all_cells()
+        if path[-1] in cells:
+            distribcell_ind = cells[path[-1]].distribcell_ind
```

Comment:
on a general note, perhaps "instance_id" or something similar be clearer than `distribcell_ind`? I'm not sure what would be most easily understandable for new developers and users. But my gut feeling is that `distribcell_ind` is more focused on preserving the legacy of the original distribcell implementation rather than forward-looking as to where the algorithm is headed in the future.